### PR TITLE
Cleanup unused usings (truly enforce ide0005 code analysis rule)

### PR DESCRIPTION
**Pull request by stan-sz**
Ensure unused `using`s are caught at the build time.

## Description

Unused `using`s are part of unused code analysis rules: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0005
Applie the suggestion from https://github.com/dotnet/roslyn/issues/41640#issuecomment-985780130 and cleaned up doc-related warnings.

## Related issue

Follows up on #3299 to clean up the repo from unused `using`s. With that in mind, the ide0005 rule being on-but-not-enforced:

- [x] I have ensured that there is a previously discussed and approved issue.

### CODE REVIEW

File: scripts/verify-nupkgs.ps1 [hunk 1]
```diff
@@ -16,7 +16,7 @@ function Verify-Nuget-Packages($packageDirectory, $version)
         "Microsoft.NET.Test.Sdk" = 16;
         "Microsoft.TestPlatform" = 603;
         "Microsoft.TestPlatform.Build" = 21;
-        "Microsoft.TestPlatform.CLI" = 477;
+        "Microsoft.TestPlatform.CLI" = 476;
```
Comment (stan-sz) [inline scripts/verify-nupkgs.ps1:19]: This file is no longer part of the nuget package: `Microsoft.TestPlatform.CLI.17.6.0-dev\contentFiles\any\netcoreapp3.1\Microsoft.TestPlatform.VsTestConsole.TranslationLayer.xml`

File: scripts/build/TestPlatform.targets [hunk 2]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <TestPlatformRoot>$(MSBuildThisFileDirectory)..\..\</TestPlatformRoot>
     <CheckEolTargetFramework>false</CheckEolTargetFramework>
-    <NoWarn>$(NoWarn);CA1416;RS0037</NoWarn>
+    <NoWarn>$(NoWarn);CA1416;RS0037;CS1570;CS1572;CS1573;CS1574;CS1584;CS1591;CS1658;CS1712;CS1734</NoWarn>
```
Comment (stan-sz) [inline scripts/build/TestPlatform.targets:6]: The cleanup of these warning suppressions (which were all ignored due to `GenerateDocumentationFile` set to `false`) is left for a separate PR.

File: src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscovererEnumerator.cs [hunk 3]
```diff
@@ -118,7 +118,6 @@ public void LoadTests(IDictionary<string, IEnumerable<string>> testExtensionSour
     /// </summary>
     /// <param name="extensionAssembly"> The extension Assembly. </param>
     /// <param name="sources"> The sources.   </param>
-    /// <param name="settings"> The settings.   </param>
```
Comment (Evangelink) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscovererEnumerator.cs:121]: This seems wrong, `settings` parameter does exist.

File: src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginManager.cs [hunk 4]
```diff
@@ -94,9 +94,6 @@ public static T CreateTestExtension<T>(Type extensionType)
     /// <typeparam name="TMetadata">
     /// Type of metadata of required extensions
     /// </typeparam>
-    /// <typeparam name="TMetadata">
```
Comment (Evangelink) [inline src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginManager.cs:97]: All these changes seem wrong, `TMetadata` type parameter does exist.

File: src/package/nuspec/TestPlatform.CLI.nuspec [hunk 5]
```diff
@@ -28,8 +28,8 @@
     <file src="licenses\LICENSE_NET.txt" target="" />
     <!-- Add a third party notice file -->
     <file src="ThirdPartyNotices.txt" target="" />
-    
+
     <!-- excluding dbghelp and cpp as they cannot be re-distributed with sdk, but need to be distributed with VS -->
-    <file src="$TargetFramework$\**\*.*" exclude="**\*.pdb;**\dbghelp.dll;**\*TestTools.CppUnitTestFramework*" target="contentFiles\any\$TargetFramework$" />
+    <file src="$TargetFramework$\**\*.*" exclude="**\*.pdb;**\*.xml;**\dbghelp.dll;**\*TestTools.CppUnitTestFramework*" target="contentFiles\any\$TargetFramework$" />
```
Comment (Evangelink) [inline src/package/nuspec/TestPlatform.CLI.nuspec:33]: Why are you excluding the xml files?

File: scripts/build/TestPlatform.targets (hunk 2 above)
Comment (Evangelink) [reply scripts/build/TestPlatform.targets:6]: Could you please create a ticket for these TODOs and also add code comment about what are these warnings.

**Review by Evangelink — COMMENTED** [review]
Hey @stan-sz, I added a couple of comments.

For future iterations, please ensure you have discussed with us or have an approved ticket before doing changes as we want to avoid wasted time/effort on changes we could reject.

File: src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginManager.cs (hunk 4 above)
Comment (stan-sz) [reply src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginManager.cs:97]: I just spotted that there was a typo in the `IMetadata`

File: src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscovererEnumerator.cs (hunk 3 above)
Comment (stan-sz) [reply src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscovererEnumerator.cs:121]: the name was incorrect

File: src/package/nuspec/TestPlatform.CLI.nuspec (hunk 5 above)
Comment (stan-sz) [reply src/package/nuspec/TestPlatform.CLI.nuspec:33]: the code analysis rule ide0005 in this repo was not enforced due to `GenerateDocumentationFile` set to `false` (see https://github.com/dotnet/roslyn/issues/41640#issuecomment-985780130). Enabling this property added enforcement of new code rules that relate to documentation annotations and generation of the output .xml doc files. If you think these files should be part of the package, I can remove the filter and update the assertion for the number of files in the nuget package accordingly.

File: scripts/build/TestPlatform.targets (hunk 2 above)
Comment (stan-sz) [reply scripts/build/TestPlatform.targets:6]: Sure. I've opened #4236

File: src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginManager.cs (hunk 4 above)
Comment (Evangelink) [reply src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginManager.cs:97]: I think `TMetadata` is correct, it's just that `IMetadata` is missing (signature is `<TPluginInfo, TExtension, IMetadata, TMetadata>`).

File: src/package/nuspec/TestPlatform.CLI.nuspec (hunk 5 above)
Comment (Evangelink) [reply src/package/nuspec/TestPlatform.CLI.nuspec:33]: I would go for including them but let's see what @MarcoRossignoli and @nohwnd think.

Comment (MarcoRossignoli) [reply src/package/nuspec/TestPlatform.CLI.nuspec:33]: We should avoid to change the layout of the packages, it's sensible. As long as we ensure that the final layout didn't change I'm fine.

Comment (stan-sz) [reply src/package/nuspec/TestPlatform.CLI.nuspec:33]: As stated in the other comment, this xml exclusion only removes `Microsoft.TestPlatform.CLI.17.6.0-dev\contentFiles\any\netcoreapp3.1\Microsoft.TestPlatform.VsTestConsole.TranslationLayer.xml` from the package (without excluding xml files, the package layout would get extra xml files due to `GenerateDocumentationFile`).

File: src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginManager.cs (hunk 4 above)
Comment (stan-sz) [reply src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginManager.cs:97]: Right, that the fix I've applied in the final interation.

**Review by Evangelink — APPROVED** [review]

**Pull request merged**